### PR TITLE
Update to 1.0, sdk 27, fix calendar bug

### LINKED ISSUE #6: Update Project for Android Oreo
To ensure that the project builds and runs on Android Oreo, the version numbers in the `build.gradle` need to be updated.
The `LifecycleActivity` extensions need to be replaced with `AppCompatActivity` in version `1.0.0` of LifeCycle Extensions.
There are changes to Room `1.0.0` that require Primary Keys to have the annotation `@NonNull`.

As well as this, Issue #2 needs to be fixed too, before the project will build.

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
Fixes #6, #2 (by lukedjpotter) and #1

### CODE REVIEW

File: app/src/main/java/com/example/android/persistence/codelab/db/Book.java [hunk 1]
```diff
@@ -18,9 +18,12 @@
...
 import android.arch.persistence.room.Entity;
 import android.arch.persistence.room.PrimaryKey;
+import android.support.annotation.NonNull;
 
 @Entity
 public class Book {
-    public @PrimaryKey String id;
+    public @PrimaryKey
+    @NonNull
+    String id;
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/db/Book.java]: do we have orders for this?
E.g.: 
```
@PrimaryKey
@NonNull
public String id;
```

Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/db/Book.java]: Done and reverted all the weirdness

File: app/build.gradle [hunk 2]
```diff
@@ -47,15 +47,18 @@ android {
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    androidTestCompile('com.android.support.test.espresso:espresso-core:' + rootProject.espressoVersion, {
+    androidTestCompile('com.android.support.test.espresso:espresso-core:3.0.1', {
```
Comment (florina-muntenescu) [inline app/build.gradle]: keep using rootProject.espressoVersion and not the direct version number here.

File: app/src/main/java/com/example/android/persistence/codelab/db/Loan.java [hunk 3]
```diff
@@ -35,7 +36,9 @@
 @TypeConverters(DateConverter.class)
 public class Loan {
     // Fields can be public or private with getters and setters.
-    public @PrimaryKey String id;
+    public @PrimaryKey
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/db/Loan.java]: Keep the order like in the `Book` class: 
``` java
@PrimaryKey
@NonNull
public String id;
```

File: app/src/main/java/com/example/android/persistence/codelab/db/User.java [hunk 4]
```diff
@@ -18,11 +18,14 @@
 
 import android.arch.persistence.room.Entity;
 import android.arch.persistence.room.PrimaryKey;
+import android.support.annotation.NonNull;
 
 
 @Entity
 public class User {
-    public @PrimaryKey String id;
+    public @PrimaryKey
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/db/User.java]: Keep the order like in the `Book` class: 
``` java
@PrimaryKey
@NonNull
public String id;
```

File: app/src/main/java/com/example/android/persistence/codelab/step3/BooksBorrowedByUserActivity.java [hunk 5]
```diff
@@ -16,31 +16,44 @@
...
-public class BooksBorrowedByUserActivity extends LifecycleActivity {
+public class BooksBorrowedByUserActivity extends AppCompatActivity {
 
     private BooksBorrowedByUserViewModel mViewModel;
 
     @SuppressWarnings("unused")
     private TextView mBooksTextView;
 
+    @SuppressWarnings("unused")
+    private static void showBooksInUi(final @NonNull List<Book> books,
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/step3/BooksBorrowedByUserActivity.java]: why is this a `private static` method? can it be just private?

File: app/build.gradle (hunk 2 above)
Comment (JoseAlcerreca) [reply app/build.gradle]: See? Merging is more work than doing it from scratch :smiley:

File: app/src/main/java/com/example/android/persistence/codelab/db/Loan.java (hunk 3 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/db/Loan.java]: I forgot to add files to the commit. Done

File: app/src/main/java/com/example/android/persistence/codelab/db/User.java (hunk 4 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/db/User.java]: Added files.

File: app/src/main/java/com/example/android/persistence/codelab/step3/BooksBorrowedByUserActivity.java (hunk 5 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/step3/BooksBorrowedByUserActivity.java]: I have absolutely no idea. Changed to non-static.

File: app/src/main/java/com/example/android/persistence/codelab/step4/TypeConvertersActivity.java [hunk 6]
```diff
@@ -16,31 +16,43 @@
...
 import java.util.List;
 
-public class TypeConvertersActivity extends LifecycleActivity {
+public class TypeConvertersActivity extends AppCompatActivity {
 
     private TypeConvertersViewModel mViewModel;
 
     private TextView mBooksTextView;
 
+    private static void showBooksInUi(final @NonNull List<Book> books,
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/step4/TypeConvertersActivity.java]: can you make this call also non-static

File: app/src/main/java/com/example/android/persistence/codelab/step4_solution/TypeConvertersActivity.java [hunk 7]
```diff
@@ -16,31 +16,43 @@
...
 import java.util.List;
 
-public class TypeConvertersActivity extends LifecycleActivity {
+public class TypeConvertersActivity extends AppCompatActivity {
 
     private TypeConvertersViewModel mViewModel;
 
     private TextView mBooksTextView;
 
+    private static void showBooksInUi(final @NonNull List<Book> books,
```
Comment (florina-muntenescu) [inline app/src/main/java/com/example/android/persistence/codelab/step4_solution/TypeConvertersActivity.java]: non static

File: app/src/main/java/com/example/android/persistence/codelab/step4/TypeConvertersActivity.java (hunk 6 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/step4/TypeConvertersActivity.java]: Done

File: app/src/main/java/com/example/android/persistence/codelab/step4_solution/TypeConvertersActivity.java (hunk 7 above)
Comment (JoseAlcerreca) [reply app/src/main/java/com/example/android/persistence/codelab/step4_solution/TypeConvertersActivity.java]: Done

**Pull request merged**